.github/scripts/print_latest_commits.py

```diff
@@ -56,7 +58,21 @@ def get_commit_results(commit: str, results: Dict[str, Any]) -> List[Dict[str, A
             )._asdict())
     return workflow_checks
 
-def isGreen(results: List[Dict[str, Any]]) -> bool:
+def isGreen(commit: str, results: Dict[str, Any]) -> bool:
+    workflow_checks = get_commit_results(commit, results)
+
+    for check in workflow_checks:
+        regex = "^linux-binary+|^android-tests+|^windows-binary+"
+        if check['workflowName'] in ["pull", "trunk", "lint"] and check['conclusion'] != 'success':
+            if check['name'] == "pull / win-vs2019-cuda11.3-py3" and check['conclusion'] == 'skipped':
+                pass
+                # there are trunk checks that run the same tests, so this pull workflow check can be skipped
+            else:
+                return False
+        elif check['workflowName'] in ["periodic", "docker-release-builds"] and check['conclusion'] not in ["success", "skipped"]:
```

Comment:
there's just a lot of if else statements in here that makes it kind of confusing. like the first and last check, why don't we just add those into the regex? 

Also maybe making it an array of regexes to match for might be easier than having 1 big regex.